### PR TITLE
feat: add server settings for default board, default color scheme and default locale

### PR DESCRIPTION
**Pull request by Meierschlumpf**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [x] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [x] Pull request targets ``dev`` branch
- [x] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

image

### CODE REVIEW

File: apps/nextjs/src/theme/color-scheme.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { cache } from "react";
+import { cookies } from "next/headers";
+
+import { db } from "@homarr/db";
+import { getServerSettingByKeyAsync } from "@homarr/db/queries";
+import type { ColorScheme } from "@homarr/definitions";
+
+export const getCurrentColorSchemeAsync = cache(async () => {
+  const cookieValue = cookies().get("homarr-color-scheme")?.value;
```
Comment (Meierschlumpf) [inline apps/nextjs/src/theme/color-scheme.ts:9]: Move key to some constant

File: packages/api/src/router/board.ts [hunk 2]
```diff
@@ -240,7 +251,20 @@ export const boardRouter = createTRPCRouter({
         })
       : null;
 
-    const boardWhere = user?.homeBoardId ? eq(boards.id, user.homeBoardId) : eq(boards.name, "home");
+    const boardSettings = await getServerSettingByKeyAsync(ctx.db, "board");
```
Comment (Meierschlumpf) [inline packages/api/src/router/board.ts:254]: Only request if not logged in / no home board set

File: packages/api/src/router/board.ts [hunk 3]
```diff
@@ -240,7 +251,20 @@ export const boardRouter = createTRPCRouter({
...
+    const boardWhere = user?.homeBoardId
+      ? eq(boards.id, user.homeBoardId)
+      : boardSettings.defaultBoardId
+        ? eq(boards.id, boardSettings.defaultBoardId)
```
Comment (Meierschlumpf) [inline packages/api/src/router/board.ts:258]: Also prevent that server default board can be made private

File: packages/translation/src/middleware.ts [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
-export const I18nMiddleware = createI18nMiddleware({
-  locales: supportedLanguages,
-  defaultLocale,
-  urlMappingStrategy: "rewrite",
-});
+export const createI18nMiddleware = (defaultLocale: SupportedLanguage) =>
+  internalCreateI18nMiddleware({
+    locales: supportedLanguages,
+    defaultLocale,
+    // TODO: check with new translation library if it uses the default or accepted language from browser
```
Comment (Meierschlumpf) [inline packages/translation/src/middleware.ts:10]: This todo will be resolved once the pr #1368 is merged to dev

File: packages/definitions/src/cookie.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export const colorSchemeCookieKey = "homarr-color-scheme";
```
Comment (manuel-rw) [inline packages/definitions/src/cookie.ts:1]:
```suggestion
export const colorSchemeCookieKey = "homarr.color-scheme";
```

File: packages/definitions/src/cookie.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
...
+export const localeCookieKey = "homarr-locale";
```
Comment (manuel-rw) [inline packages/definitions/src/cookie.ts:2]:
```suggestion
export const localeCookieKey = "homarr.locale";
```

**Review by manuel-rw — DISMISSED** [review]

**Comment by Meierschlumpf** [comment]
Test issue: https://github.com/vitest-dev/vitest/issues/5101
Fixing it right now by adding default coverage excludes to `vitest.config.ts`

**Review by manuel-rw — APPROVED** [review]
lgtm

**Pull request merged**